lib/new_relic/traced_thread.rb

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  #
+  # This class allows the state of a transaction to be passed to a Thread so that nested segments can be created from the operations performed within the Thread's block.
+  # To have the New Relic Ruby agent automatically trace all of your applications threads,
+  # enable the `instrumentation.thread.tracing` configuration option in your newrelic.yml.
+  #
+  # Note: disabling the configuration option `instrumentation.thread` while using this class can cause incorrectly nested spans.
+  #
+  # @api public
+  class TracedThread < Thread
```

Comment:
Grammatically I'm wondering if this is "TracedThread" then the other name should be "MonitoredThread" or if that one should stay "ThreadMonitor" and this one because "ThreadTrace". But given that they are used in an either / or fashion, it's not a blocker.

Reply:
I like TracedThread over ThreadTrace. Since this inherits from Thread and adds tracing in it, i think TracedThread seems more obvious that it's creating a thread object. As for ThreadMonitor, I have no preference on that name really because originally it was just Thread, but that was causing annoying namespace issues in other parts of the agent so I just changed it to anything else to avoid that. I'd be down to update that one.

Reply:
Ok cool. Let's called TracedThread for sure. @kaylareopelle any preference on the naming of `ThreadMonitor`? We can name it anything but `Thread`.

Reply:
I think I'd prefer MonitoredThread over ThreadMonitor :) 
Another option could be InstrumentedThread, given that it's the instrumentation for the Thread. I know that's a bit redundant with the prefix, but I kind of like how `NewRelic::Agent::Instrumentation::InstrumentedThread` reads. Thoughts?

Reply:
Hmm i'm less of a fan of that, the double 'Instrument' part looks odd to me. personally, I like MonitoredThread over InstrumentedThread i think.

Reply:
Cool! that works!